### ISSUE TITLE
set_xticklabels() is not working properly for factorplot() if there's a col_wrap

### ISSUE THREAD

**Issue by jdoepfert** [mwaskom/seaborn#1126]
`set_xticklabels()` is not working properly for `factorplot()` if there's a col_wrap happening, and `sharex=False`.

Example:
```
import pandas as pd
import seaborn as sns
df = pd.DataFrame({'num': [1, 2, 3, 4, 5, 6], 
                   'abc': ['a', 'a', 'b', 'b', 'c', 'c'],
                   'xy': ['x', 'x', 'x', 'y', 'y', 'y']})
g = sns.factorplot(data=df, x='xy', y='num', col='abc', kind='bar', col_wrap=2, sharex=False)
g.set_xticklabels(rotation=90)
```

In the resulting figure, the axis labels have not been rotated for the upper left plot: 
![image](https://cloud.githubusercontent.com/assets/7914340/23343333/ff43ff28-fc69-11e6-9f36-55341ead7dc0.png)

**Comment by jdoepfert** [comment]
I found this workaround:
Instead of `g.set_xticklabels(rotation=90)`, use
```
for ax in g.axes.flatten(): 
    for tick in ax.get_xticklabels(): 
        tick.set(rotation=30)
```
From http://stackoverflow.com/questions/26540035/rotate-label-text-in-seaborn-factorplot

**Comment by mwaskom** [comment]
This issue is not `col_wrap` really, it is that `sharex=False`. `set_xticklabels(rotation=90)` assumes only the "outer" ticks labels are shown, which is normally the case.

**Comment by jdoepfert** [comment]
OK got it thx! Would be nice if it would rotate all the tick labels, though :)

**Comment by mwaskom** [comment]
Lots of things "would be nice". This is a low priority for me. IMO one should almost always be sharing axes when using `FacetGrid`.

**Comment by mwaskom** [comment]
Closed with #1313
